internal/bminventory/inventory.go

```diff
@@ -1519,6 +1519,13 @@ func (b *bareMetalInventory) refreshClusterHosts(ctx context.Context, cluster *c
 				h.ID.String(), cluster.ID.String())
 			return common.NewApiError(http.StatusNotFound, err)
 		}
+
+		// Refresh inventory - especially disk eligibility. The host requirements might have changed.
+		err = b.refreshInventory(ctx, cluster, h, tx)
```

Comment:
How often is `refreshClusterHosts` called? Maybe there can be performance implications for unmarshaling & marshaling the huge inventory JSON for all hosts here?

@filanov what do you think?

Reply:
That methods is called from each `UpdateClusterInternal` call, which can come from four places:
- manual cluster update;
- reconciliation of objects only when cluster update is required:
  -  `InfraEnv`;
  -  `ClusterDeployment`;
  -  `Agent`;

The manual one should not be an issue, as it is a low-rate request type; the reconciliation related ones can potentially be more frequent.

Reply:
As @jakub-dzon said controller calling cluster update only when needed, controller fetches the data from the DB compare and update if needed. 
Update inventory is called once a minute for each host. before and during the installation. so for now I don't think that we will have performance issues.